tests/foreman/cli/test_registration.py

```diff
@@ -224,3 +226,42 @@ def test_negative_global_registration_without_ak(module_target_sat):
         'Failed to generate registration command:\n  Missing activation key!'
         in context.value.message
     )
+
+
+def test_positive_custom_facts_for_host_registration(
+    module_sca_manifest_org,
+    module_location,
+    module_target_sat,
+    rhel8_contenthost,
+    module_activation_key,
+):
+    """Attempt to register a host and check all the interfaces are created from the custom facts
+
+    :id: db73c146-4557-4bf4-a8e2-950ecba31620
+
+    :steps:
+        1. Register the host.
+        2. Check the host is registered and all the interfaces are created successfully.
+
+    :expectedresults: Host registered successfully with all interfaces created from the custom facts.
+
+    :BZ: 2250397
+
+    :customerscenario: true
+    """
+    facts = {
+        f"net.interface.{gen_string('alphanumeric')}.mac_address": gen_mac(),
+        f"net.interface.{gen_string('alpha')}.{gen_string('numeric')}.mac_address": gen_mac(),
+        f"net.interface.{gen_string('alpha')}.{gen_string('numeric')}.mac_address": gen_mac(),
+    }
+    _, facts_file = mkstemp(suffix='.facts')
+    with open(facts_file, "w") as f:
+        json.dump(facts, f, indent=4)
+    rhel8_contenthost.put(facts_file, '/etc/rhsm/facts/')
+    org = module_sca_manifest_org
+    result = rhel8_contenthost.register(
+        org, module_location, [module_activation_key.name], module_target_sat
+    )
+    assert result.status == 0, f'Failed to register host: {result.stderr}'
+    host_info = module_target_sat.cli.Host.info({'name': rhel8_contenthost.hostname})
+    assert len(host_info['network-interfaces']) == len(list(facts)) + 1
```

Comment:
do we have a default interface + facts here, right? could you add a comment here?
And is it possible to add more assertions to verify network-interfaces from available facts?